dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -128,13 +136,28 @@ public T createHosted(@ApiParam(hidden = true) @Auth User user,
         long l = getEntryDAO().create(entry);
         T byId = getEntryDAO().findById(l);
         elasticManager.handleIndexUpdate(byId, ElasticMode.UPDATE);
-        return byId;
+
+        URI requestURI = uriInfo.getRequestUri();
+        int heIndex = requestURI.toString().indexOf("/hostedEntry");
+        String uriStrPrefix = requestURI.toString().substring(0, heIndex);
+        String uriStr = uriStrPrefix + "/" + Long.toString(l);
+
+        URI uriEntity = null;
+        try {
+            uriEntity = new URI(uriStr);
+        } catch (URISyntaxException e) {
+            LOG.error("URI for hosted workflow or tool is malformed");
+        }
+
+        // If uriEntity is null the Response created call will still work
+        return Response.created(uriEntity).entity(byId).build();
```

Comment:
Seems clunky.  I think the Dropwizard convention is to return domain objects if possible 
https://www.dropwizard.io/1.0.5/docs/manual/core.html#responses

I can see #1859 but what specific problem was this causing?

Reply:
When I was writing tests in Gatling, I was checking for a 201 status code to indicate success, as that is the standard response code for a successful POST.

But it was easy enough to change the Gatling code to check for a 200.

We would only need to address this from a "purist" RESTful perspective, but there are no concrete problems that this will solve that I'm aware of. And if most clients are using Swagger generated client code rather than doing the http calls directly, it really won't matter.

I'm OK closing this and the corresponding ticket.

Reply:
I think it's ok to leave the ticket in case we come across a cleaner way of accomplishing this. 
But as is, I think the decrease in readability outweighs the benefit of conforming more closely to REST principles.

Reply:
OK I understand; I will close the PR.